lib/log_vty.c

```diff
@@ -459,6 +460,70 @@ DEFUN (clear_log_cmdline,
 	return CMD_SUCCESS;
 }
 
+/* Per-daemon log file config */
+DEFUN (config_log_dmn_file,
+       config_log_dmn_file_cmd,
+       "log daemon " DAEMONS_LIST " file FILENAME [<emergencies|alerts|critical|errors|warnings|notifications|informational|debugging>$levelarg]",
+       "Logging control\n"
+       "Specific daemon\n"
+       DAEMONS_STR
+       "Logging to file\n"
+       "Logging filename\n"
+       LOG_LEVEL_DESC)
+{
+	int level = log_default_lvl;
+	int idx = 0;
+	const char *d_str;
+	const char *filename;
+	const char *levelarg = NULL;
+
+	d_str = argv[2]->text;
+
+	/* Ignore if not for this daemon */
+	if (!strmatch(d_str, frr_get_progname()))
+		return CMD_SUCCESS;
+
+	if (argv_find(argv, argc, "file", &idx))
+		filename = argv[idx + 1]->arg;
+	else
+		return CMD_SUCCESS;
+
+	if (argc > 5)
+		levelarg = argv[5]->text;
+
+	if (levelarg) {
+		level = log_level_match(levelarg);
+		if (level == ZLOG_DISABLED)
+			return CMD_ERR_NO_MATCH;
+	}
+	return set_log_file(&zt_file, vty, filename, level);
```

Comment:
So I believe this is going to replace the general non-daemon specific log file. I don't think that's the behavior we want though.

Currently when one runs the daemon with a `--log` CLI arg the daemon  logs to that file (and no-one else does); however, when you configure the normal logfile (i.e., not a per-daemon) the daemon logs to that shared log file *additionally* (i.e., it doesn't replace the earlier logging).

This is b/c the 2 log files are using different `struct zlog_cfg_file *` "targets". The config based one is using `zt_file` target, the CLI arg one is using `zt_file_cmdline`. I think we want the per daemon log file to use the `zt_file_cmdline` target (we could rename it zt_per_daemon_file if we we don't want the confusion of calling it cmdline).

Reply:
that's not what I tried to do - I didn't try to replace the existing command-line option, which as you say has its own infra in place. I was only trying to offer a way to support per-daemon log files in the presence of integrated config.

if someone wants to get rid of the command-line option, that's, IMO, a different discussion, not gated by the config consolidation.

Reply:
OK so this is equivalent to having a log command inside a per-daemon config file then? That seems like the right thing to do then I agree.

Do you know what happens if you pick the same filename as the one given on the command line when the daemon is launched?

Reply:
> Do you know what happens if you pick the same filename [...]

just tried that out: the config gets an error, though it's accepted: the file can't be opened a second time by the same process, it looks like.